src/rest_vol_link.c

```diff
@@ -120,6 +121,12 @@ RV_link_create(H5VL_link_create_args_t *args, void *obj, const H5VL_loc_params_t
             new_link_loc_obj = (RV_object_t *)hard_link_target_obj;
     } /* end if */
 
+    if (new_link_loc_obj && ((base_URL = new_link_loc_obj->domain->u.file.server_info.base_URL) == NULL))
```

Comment:
These conditions need to be split so that assigning of `base_URL` doesn't depend on `new_link_loc_obj` being non-NULL